edc-extensions/hashicorp-vault/src/main/java/org/eclipse/tractusx/edc/hashicorpvault/AbstractHashicorpVaultExtension.java

```diff
@@ -50,6 +51,14 @@ public class AbstractHashicorpVaultExtension {
 
     private static final String VAULT_TIMEOUT_SECONDS = "edc.vault.hashicorp.timeout.seconds";
 
+    public HashicorpVaultClient createVaultClient(ServiceExtensionContext context, ObjectMapper mapper) {
```

Comment:
should be `protected`, maybe even a lazy-init singleton, so that both inheritors use the same instance of the vault client?